backend/tests/models/services/account.test.ts

```diff
@@ -0,0 +1,668 @@
+/*
+ * @jest-environment node
+ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as AccountServices from '../../../src/models/services/account.js';
+import { Config, Entry, Journal, User } from '../../../src/models/index.js';
+import mongoose, { HydratedDocument } from 'mongoose';
+import { ConfigType } from '../../../src/models/config.js';
+import ExpressError from '../../../src/utils/ExpressError.js';
+import { JournalType } from '../../../src/models/journal.js';
+import { UserType } from '../../../src/models/user.js';
+import connectDB from '../../../src/db.js';
+
+describe('Account services tests', () => {
+  let mockUser: UserType;
+
+  beforeAll(async () => {
+    await connectDB('cdj');
+  });
+
+  beforeEach(async () => {
+    mockUser = await User.register({
+      fname: 'test fname',
+      lname: 'test lname',
+      email: 'testemail@fake.com',
+    } as UserType, 'OldPassword123!');
+  });
+
+  afterEach(async () => {
+    await mongoose.connection.dropDatabase();
+    jest.clearAllMocks();
+  });
+
+  afterAll(async () => {
+    await mongoose.disconnect();
+  });
+
+  describe('getAccount tests', () => {
+    it('gets user and config without message for getAccount when config set', async () => {
+      const mockConfig = await Config.create({
+        model: {
+          chat: 'test chat model',
+          analysis: 'test analysis model'
+        }
+      });
+      const mockJournal = await Journal.create({ 
```

Comment:
Similar to `User` and `Config`, this is an actual instance of `Journal`.


```suggestion
      const testJournal = await Journal.create({ 
```

Reply:
I see that there are multiple tests (not all) with this naming mixup. Should be easy to to update the names with find+replace. In later tests below, I notice "mock" is no longer used but simply `const journal = await Journal.create({ ... )};` like in the test `should handle journals with no entries gracefully`. So either `testJournal` or simply `journal` should be used. Same can be applied to `config` and `user`.